Modules/posixmodule.c

```diff
@@ -7858,9 +7864,9 @@ os_fork1_impl(PyObject *module)
         /* child: this clobbers and resets the import lock. */
         PyOS_AfterFork_Child();
     } else {
-        warn_about_fork_with_threads("fork1");
         /* parent: release the import lock. */
         PyOS_AfterFork_Parent();
+        warn_about_fork_with_threads("fork1");
```

Comment:
since it isn't entirely obvious to a reader why this needs to be called at this point instead of earlier, could you add a comment about requirements to `warn_about_fork_with_threads` itself and ideally if possible an assertion that the interpreter is in a desirable state for it to be called.  _(otherwise it could become a pain in future refactorings if one of these were to be moved to be called in a wrong spot)_

Reply:
I've added a comment and assertion to `warn_about_fork_with_threads`.